Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -225,7 +302,29 @@ public struct HTTPHeaders: CustomStringConvertible {
     ///
     /// - Parameter name: The name of the header field to remove from the block.
     public mutating func remove(name: String) {
-        self.storage[name.lowercased()] = nil
+        guard !self.headers.isEmpty else {
+            return
+        }
+
+        let utf8 = name.utf8
+        let stringLength = utf8.count
+        var array: [Int] = []
+        // We scan from the back to the front so we can remove the subranges with as less overhead as possible.
+        for idx in stride(from: self.headers.count - 1, to: -1, by: -1) {
+            let header = self.headers[idx]
+            if stringLength == header.name.length && self.buffer.equalCaseInsensitiveASCII(seq: utf8, at: header.name) {
+                array.append(idx)
+            }
+        }
+
+        guard !array.isEmpty else {
+            return
+        }
+
+        array.forEach {
+            self.headers.remove(at: $0)
+        }
+        self.continuous = false
```

Comment:
So, at the cost of a bit more computation we can improve our writing performance here, though I don't know how useful that is.

Specifically, if we remove a contiguous block of headers either at the start or the end of the byte buffer, we don't need to unset `continuous`. Doing this requires a quick check on the indices, but it probably isn't that hard to code it up, and it will save us an increased cost writing the header block out.

Not sure how useful it is, just worth thinking about.

Probably worth doing as it's own PR so we can benchmark the computational effect of it.